### ISSUE TITLE
Goal and Update of nvidia/caffe

### ISSUE THREAD

**Issue by hevy-CC4477** [NVIDIA/caffe#74]
Hi,

I have 2 questions:
- What is the goal of this fork ? (Is it to furnish stable version of BVLC/caffe ?)
- If yes, what is your merge strategic ? What I mean is: How do you choose when you merge the master on this fork ? (actually ~3 months late)

Thanks,

**Comment by bhack** [comment]
:+1:

**Comment by lukeyeager** [comment]
NVcaffe releases and development coincide with our [Deep Learning Software](https://developer.nvidia.com/deep-learning-software) releases. When we start preparing for a new release, we create a new release branch from BVLC’s `master` branch to pull in the newest features, and then start developing our own features and doing a bunch of stability testing.  During this phase, we may merge bugfixes from the BVLC fork (examples: https://github.com/NVIDIA/caffe/pull/44, https://github.com/NVIDIA/caffe/pull/79) and submit bugfixes we find back to BVLC (example: https://github.com/BVLC/caffe/pull/3574). Because we are packaging components into debs, we need stable forks and versioning. https://github.com/BVLC/caffe/pull/3311 was a step in the right direction in terms of getting versioning into Caffe, but we also need release management and scheduling for test coverage.
 
In short, BVLC doesn’t have the same release engineering concerns that we do and sometimes cannot take in patches for new functionality that align with our release schedules like cuDNN updates, CUDA toolkit changes, etc.  Our goal is to push our work and bugfixes into the main tree as soon as viable.
 
We also have experimental work that is far from being mergeable into BVLC `master`, like our fp16 work, which allows us to expose research work to the community without the large noise and churn of a PR that might be long standing.

**Comment by lukeyeager** [comment]
Sorry for the late response @hevy-CC4477 - hopefully that answers your question!
